libcxx/include/istream

```diff
@@ -292,6 +294,10 @@ public:
   basic_istream& getline(char_type* __s, streamsize __n, char_type __dlm);
 
   basic_istream& ignore(streamsize __n = 1, int_type __dlm = traits_type::eof());
+  template <class _Tp = char_type, __enable_if_t<is_same<_Tp, char_type>::value, int> = 0>
```

Comment:
Currently this overload is enabled for every `char_type`. It should only be enabled when `char_type` is `char`.
```suggestion
  template <class _Tp = char_type, __enable_if_t<is_same<_Tp, char>::value, int> = 0>
```

It's curious that no existing test case for other `char_type` is affected... No change requested for additional coverage in this PR.

Reply:
Thank you for noticing that.